File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetDeviceGroupPresentCapabilitiesKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDeviceGroupPresentCapabilitiesKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pDeviceGroupPresentCapabilities, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceGroupPresentCapabilitiesKHR f = (PFN_vkGetDeviceGroupPresentCapabilitiesKHR)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDeviceGroupPresentCapabilitiesKHR*)jniLongToPointer(pDeviceGroupPresentCapabilities))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetDeviceGroupPresentCapabilitiesKHR_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pDeviceGroupPresentCapabilities){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceGroupPresentCapabilitiesKHR f = (PFN_vkGetDeviceGroupPresentCapabilitiesKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDeviceGroupPresentCapabilitiesKHR*)jniLongToPointer(pDeviceGroupPresentCapabilities))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkGetDeviceGroupPresentCapabilitiesKHR.java
package cz.mg.vulkan;

public class PFNvkGetDeviceGroupPresentCapabilitiesKHR extends VkFunctionPointer {
    public PFNvkGetDeviceGroupPresentCapabilitiesKHR() {
    }

    protected PFNvkGetDeviceGroupPresentCapabilitiesKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetDeviceGroupPresentCapabilitiesKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDeviceGroupPresentCapabilitiesKHR(long value) {
        setValue(value);
    }

    public PFNvkGetDeviceGroupPresentCapabilitiesKHR(VkInstance instance) {
        super(instance, new VkString("vkGetDeviceGroupPresentCapabilitiesKHR"));
    }

    public void call(VkDevice device, VkDeviceGroupPresentCapabilitiesKHR pDeviceGroupPresentCapabilities, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pDeviceGroupPresentCapabilities != null ? pDeviceGroupPresentCapabilities.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long pDeviceGroupPresentCapabilities, long rval);


    public int call(VkDevice device, VkDeviceGroupPresentCapabilitiesKHR pDeviceGroupPresentCapabilities){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pDeviceGroupPresentCapabilities != null ? pDeviceGroupPresentCapabilities.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long pDeviceGroupPresentCapabilities);

}
